src/wallet/rpcwallet.cpp

```diff
@@ -3660,7 +3660,11 @@ UniValue createrawpegin(const JSONRPCRequest& request)
     stack.push_back(txData);
     stack.push_back(txOutProofData);
 
-    if (!IsValidPeginWitness(pegin_witness, mtx.vin[0].prevout)) {
+    // Peg-in witness isn't valid, even though the block header is(without depth check)
+    // We re-check depth before returning with more descriptive result
+    if (GetBoolArg("-validatepegin", DEFAULT_VALIDATE_PEGIN) &&
```

Comment:
How about:
```
if (GetBoolArg("-validatepegin", DEFAULT_VALIDATE_PEGIN) {

            if (!IsConfirmedBitcoinBlock(merkleBlock.header.GetHash(), 0)) {
                 throw JSONRPCError(RPC_INVALID_PARAMETER, "Parent chain block not mature.");
            }
            if (!IsValidPeginWitness(pegin_witness, mtx.vin[0].prevout)) {
                 throw JSONRPCError(RPC_INVALID_PARAMETER, "Constructed peg-in witness is invalid.");
            }
     }
```

Reply:
we need to pass back a "maturity" field, so we don't want to throw an error there.

Leaving as-is for now.